src/main/UserActivityMonitor.js

```diff
@@ -0,0 +1,148 @@
+  }
+  get userIdleTime() {
+    return this.idleTime;
+  }
+
+  // class methods
+  // - begin monitoring system events and idle time at defined frequency
+  startMonitoring(config = {}) {
+    if (app.isReady()) {
+      // store config
+      this.config = Object.assign({}, this.config, config);
+
+      // handle system events
+      electron.powerMonitor.on('suspend', this.handleSuspend);
+      electron.powerMonitor.on('resume', this.handleResume);
+      electron.powerMonitor.on('shutdown', this.handleShutdown);
+      electron.powerMonitor.on('lock-screen', this.handleLockScreen);
+      electron.powerMonitor.on('unlock-screen', this.handleUnlockScreen);
+
+      // monitor system idle time
+      clearInterval(this.systemIdleTimeIntervalID);
+      this.systemIdleTimeIntervalID = setInterval(() => {
+        try {
+          electron.powerMonitor.querySystemIdleTime((idleTime) => {
+            this.updateIdleTime(idleTime);
+          });
+        } catch (err) {
+          console.log('Error getting system idle time:', err);
+        }
+      }, this.config.regularUpdateFrequency);
+    } else {
+      this.emit('error', new Error('UserActivityMonitor.startMonitoring can only be called after app is ready'));
+    }
+  }
+
+  // - stop monitoring system events and idle time
+  stopMonitoring() {
+    // stop monitoring system events
+    electron.powerMonitor.off('suspend', this.handleSuspend);
+    electron.powerMonitor.off('resume', this.handleResume);
+    electron.powerMonitor.off('shutdown', this.handleShutdown);
+    electron.powerMonitor.off('lock-screen', this.handleLockScreen);
+    electron.powerMonitor.off('unlock-screen', this.handleUnlockScreen);
+
+    // stop monitoring system idle time
+    clearInterval(this.systemIdleTimeIntervalID);
+  }
+
+  // - updates internal idle time properties and conditionally triggers various tasks based on those values
+  updateIdleTime(idleTime) {
+    // store last and this idleTime for comparison
+    this.lastIdleTime = this.idleTime;
+    this.idleTime = idleTime;
+
+    // provide frequent activity updates
+    this.emit('update', {userIsActive: this.isActive, idleTime: this.idleTime});
+
+    if (this.isActive && this.idleTime > this.config.userActivityTimeout) {
+      // update activity status to inactive after defined time
+      this.updateUserActivityStatus(false);
+    } else if (!this.isActive && !this.forceUserIsInactive && this.idleTime < this.config.userActivityTimeout && this.lastIdleTime > this.idleTime) {
+      // update activity status to active if system idle time resets
+      this.updateUserActivityStatus(true);
+    } else if (new Date().getTime() - this.lastStatusUpdate > this.config.statusUpdateThreshold) {
+      // provide periodic status updates
+      this.sendStatusUpdate();
+    }
+  }
+
+  // - updates user activity status if changed and triggers a status update
+  updateUserActivityStatus(userIsActive = false) {
```

Comment:
nit: parameter here could just be `isActive`, and it won't conflict with `this.isActive`.